ide/coqide/dune

```diff
@@ -27,13 +27,17 @@
  (name coqide_gui)
  (wrapped false)
  (modules (:standard \ document idetop coqide_main default_bindings_src))
+ (foreign_stubs
+  (language c)
+  (names coqide_os_stubs))
  (optional)
  (libraries coqide-server.protocol coqide-server.core lablgtk3-sourceview3))
 
-(rule
- (targets coqide_os_specific.ml)
- (deps (:in-file coqide_X11.ml.in)) ; TODO support others
- (action (copy# %{in-file} %{targets})))
+(rule (targets coqide_os_specific.ml)
+ (action (copy# coqide_%{read:../../config/dune.gtk_platform}.ml.in %{targets})))
```

Comment:
Changed to use this copy method (allowing to use `copy#` for the ml file)
Testing says that dune understands the dependencies for `(action (copy#))` even though `%{read}` isn't allowed if we try to make it explicit with a `(deps)`.

Reply:
Umm, in principle that should work but I dunno if we are walking a bit of an untested path here.

Dune 3.0 will support `%{read}` and `:include` everywhere [including deps], but for the 2.0 series, there are still some restrictions regarding dynamic build rules so I dunno; I guess we will see if this works or gives some trouble.

Reply:
In fact, in 3.0 we could just have configure generate the stanza itself, and use include.

Reply:
Another downside of this methods [which is not a big deal but should be pointed out] is that splitting coqide to its own package becomes harder; as we now have to take care of the config file.

Using ml libraries to manage this stuff allows us to reuse more infrastructure.

Reply:
Yeah, coqide packages now have to run configure, I don't like that.

Reply:
After thinking about this, please move the file generation from `./configure` to a small binary in `ide/coqide`.